elasticdl/python/common/k8s_job_monitor.py

```diff
@@ -0,0 +1,172 @@
+import time
+
+from elasticdl.python.common.constants import PodStatus
+from elasticdl.python.common.k8s_client import Client
+from elasticdl.python.common.log_utils import default_logger as logger
+
+MAX_READ_POD_RETRIES = 6
+
+
+def print_tail_log(log, tail_num):
+    if log is not None:
+        log_lines = log.split("\n")
+        tail_index = -1 * tail_num
+        logger.info("\n".join(log_lines[tail_index:]))
+
+
+class PodMonitor:
+    def __init__(self, namespace, pod_name, use_kube_config=True):
+        """
+        k8s Pod Monitor. During data transformation, we may launch a pod
+        on k8s cluster to perform data analysis and need to monitor the
+        pod execution status.
+
+        Args:
+            namespace: The name of the Kubernetes namespace where the pod
+                will be created.
+            pod_name: Pod name, should be unique in the namespace.
+            use_kube_config: If true, load the cluster config from
+                ~/.kube/config. Otherwise, if it's in a process running in
+                a K8S environment, it loads the incluster config.
+        """
+        self.namespace = namespace
+        self.pod_name = pod_name
+        self.client = Client(
+            image_name=None,
+            namespace=namespace,
+            job_name=None,
+            force_use_kube_config_file=use_kube_config,
+        )
+
+    def monitor_status(self):
+        retry_num = 0
+        pod_succeeded = False
+
+        while True:
+            pod = self.client.get_pod(self.pod_name)
+            if pod is None:
+                retry_num += 1
+                if retry_num > MAX_READ_POD_RETRIES:
+                    logger.error("{} Not Found".format(self.pod_name))
+                    break
+                time.sleep(10)
+                continue
+
+            logger.info("Pod Status : %s" % pod.status.phase)
+            if pod.status.phase == PodStatus.SUCCEEDED:
+                pod_succeeded = True
+                break
+            elif pod.status.phase == PodStatus.FAILED:
+                logger.info(self.client.get_pod_log(self.pod_name))
+                break
+            else:
+                time.sleep(30)
+        self.client.delete_pod(self.pod_name)
```

Comment:
This method does not only monitor the pod status, but it also deletes the pod. May rename the method name to reflect this.

Reply:
Split the method to two methods monitor and delete?